tflint/annotation.go

```diff
@@ -40,7 +40,7 @@ func NewAnnotations(path string, file *hcl.File) (Annotations, hcl.Diagnostics)
 			continue
 		}
 		ret = append(ret, Annotation{
-			Content: match[1],
+			Content: strings.TrimSpace(match[1]),
```

Comment:
Note: The `TrimSpace` is called internally in `IsAffected`, so this change probably won't affect the behavior.
https://github.com/terraform-linters/tflint/blob/8eecbda45d83f75594932840ed1892a3e571ddcc/tflint/annotation.go#L59

But I think it's good practice to pre-remove extra spaces here.

Reply:
If I were to remove this trimming here, the tests would need to have extra spaces at the end of the `Content:` fields, which to me is counter intuitive, and easy to miss.
I think it is better to clean the `Content` as early as we can.

You also wrote that 
> But I think it's good practice to pre-remove extra spaces here.

so it's ok to leave this change in?

---

This is how the tests would have to look without the trimming:

```patch
Index: tflint/annotation_test.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tflint/annotation_test.go b/tflint/annotation_test.go
--- a/tflint/annotation_test.go	(revision 5f5f91283906530c0cc3a82f0d4ea9d41248ab21)
+++ b/tflint/annotation_test.go	(date 1698227877022)
@@ -33,7 +33,7 @@
 
 	expected := Annotations{
 		{
-			Content: "aws_instance_invalid_type, terraform_deprecated_syntax",
+			Content: "aws_instance_invalid_type, terraform_deprecated_syntax ",
 			Token: hclsyntax.Token{
 				Type:  hclsyntax.TokenComment,
 				Bytes: []byte("/* tflint-ignore: aws_instance_invalid_type, terraform_deprecated_syntax */"),
@@ -69,7 +69,7 @@
 			},
 		},
 		{
-			Content: "aws_instance_invalid_type",
+			Content: "aws_instance_invalid_type ",
 			Token: hclsyntax.Token{
 				Type:  hclsyntax.TokenComment,
 				Bytes: []byte("// tflint-ignore: aws_instance_invalid_type // With reason\n"),
@@ -81,7 +81,7 @@
 			},
 		},
 		{
-			Content: "aws_instance_invalid_type",
+			Content: "aws_instance_invalid_type ",
 			Token: hclsyntax.Token{
 				Type:  hclsyntax.TokenComment,
 				Bytes: []byte("# tflint-ignore: aws_instance_invalid_type # With reason\n"),
```

Reply:
> so it's ok to leave this change in?

Yes!